src/FiniteElement.m

```diff
@@ -1404,13 +1440,29 @@ function createComplementarityConstraints(obj, p_stage)
             cross_comp_pairs = obj.getCrossCompPairs();
 
             cross_comp_aggregated = [];
+            ind_std_comp = [];
 
             sigma_scale = 1; % TODO scale properly
                              % apply psi
             g_cross_comp = [];
             lbg_cross_comp = [];
             ubg_cross_comp = [];
             if problem_options.cross_comp_mode == CrossCompMode.STAGE_STAGE || ~problem_options.use_fesd
+                % NOTE: this is a hack.
```

Comment:
This is the "working" version of the `getStdCompPairs` method (which should be removed). Essentially  it is trying to populate an index set into the complementarity functions G and H that gets only the standard complementarities from the DCS discretization. I think in the long run this can be removed as we don't want to support the Stage-Stage complementarity mode.

Reply:
The indexing starting from 3 for CLS is a horrible hack that came from quickly jamming that into a design that was not well equipped to be extended :sweat_smile: